pkg/reconciler/configmappropagation/configmappropagation.go

```diff
@@ -341,3 +342,14 @@ func (r *Reconciler) contains(name string, list []*corev1.ConfigMap) (*corev1.Co
 	}
 	return nil, false
 }
+
+// removeOwnerReference removes the target ownerReference and returns a new slice of ownerReferences.
+func (r *Reconciler) removeOwnerReference(ownerReferences []metav1.OwnerReference, uid types.UID) []metav1.OwnerReference {
+	var expected []metav1.OwnerReference
+	for _, owner := range ownerReferences {
+		if owner.UID != uid {
+			expected = append(expected, *owner.DeepCopy())
```

Comment:
Since the object itself isn't being modified, just the slice, I'd expect `append(expected, owner)` to work fine here. Why use DeepCopy?

Reply:
done